### ISSUE TITLE
AKSamplePlayer crashes in the simulator

### ISSUE THREAD

**Issue by knagle** [AudioKit/AudioKit#1100]
Using the Recorder iOS example I can build and run the project in both the simulator and on any iOS device with no issues. However, if I replace the AKAudioPlayer in the project with an AKSamplePlayer the project runs on iOS devices but crashes in the simulator. Even if I just initialise an AKSamplePlayer object with a valid file but do nothing else, it crashes on initialisation, pointing to a problem in ftbl.c in sp_ftbl_init

at line 18: 

ft->lobits = log2(SP_FT_MAXLEN / size); 

I get the error below.

Thread 1: EXC_ARITHMETIC (code=EXC_I386_DIV, subcode=0x0)

A division by zero error? Where size is 0?

This doesn't occur on an iOS device. Just the simulator. The same problem occurs in my own project, which makes extensive use of AKSamplePlayer. 

Any idea why it wouldn't work in the simulator but work fine on any and all iOS devices tested?

**Comment by aure** [comment]
THanks for pointing this out.  My notification of new issues was broken so I didn't see this until now, but will look into it.

**Comment by eljeff** [comment]
Can you share your code where you replaced the player with a sampleplayer? They are pretty different so I'd like to see how they're being built etc.

**Comment by aure** [comment]
Closed until further information received.

**Comment by knagle** [comment]
Below is the only modification I make to the Recorder iOS demo app to get a crash on the simulator. ViewController.swift line 70 onward.

`// Will set the level of microphone monitoring`
`micBooster.gain = 0`
`recorder = try? AKNodeRecorder(node: micMixer)`
`if let file = recorder.audioFile {`
`// Here we're just trying to initialise a sample player object with a file`
`let samplePlayer = AKSamplePlayer(file: file)`
`// original code`
`player = try? AKAudioPlayer(file: file)`
`}`
`player.looping = true`

Apologies for not sharing this code sooner.

**Comment by aure** [comment]
What's happening in your example is that the file that is being told to be loaded is nil.  We've taken a few liberties with this example for brevity, but really there should not be any forced unwrapping and in the case of sample player, better off starting with nothing and then loading data once you're already sure something has been recorded.  I have fixed the example while researching this question though, so thank you for that.  In general we should provide a better recorder example and I've added it to our projects list:  https://github.com/AudioKit/AudioKit/projects/4
